# Homework Two: Linear Programs

### Question One

Story Problem : Farmer Jane owns 45 acres of land. She is going to plant each with wheat or corn. Each acre planted with wheat yields 230 profit; each with corn yields 190 profit. The labor and fertilizer used for each acre are given in the table below. 100 workers and 130 tons of fertilizer areavailable.

## Expanded solution
#### Problem Data

In [1]:
using JuMP, Clp

m = Model(solver = ClpSolver())

@variable(m, wheat >= 0)                  # wheat
@variable(m, corn >= 0)                   # corn

@constraint(m, wheat + corn <= 45)        # total land to work on
@constraint(m, 3wheat + 2corn <= 100)     # total number of labor
@constraint(m, 2wheat + 4corn <= 130)     # total amount of fertilizer

@objective(m, Max, 230wheat + 190corn)    # maximize profit

m

Maximization problem with:
 * 3 linear constraints
 * 2 variables
Solver is Clp

#### Problem Model

In [2]:
println(m)
status = solve(m)
println(status)

Max 230 wheat + 190 corn
Subject to
 wheat + corn ≤ 45
 3 wheat + 2 corn ≤ 100
 2 wheat + 4 corn ≤ 130
 wheat ≥ 0
 corn ≥ 0

Optimal


In [3]:
using JuMP
println("Acres of Wheat:" , getvalue(wheat))
println("Acres of Corn: " , getvalue(corn))

profit = (230 * wheat) + (190 * corn)
println("Total profit: \$", getvalue(profit))

Acres of Wheat:17.499999999999993
Acres of Corn: 23.750000000000004
Total profit: $8537.5


##### The ideal solution, and one I would recommend, is to cover 17.5 acres with Wheat and 23.75 acres of Corn. This will produce the optimal profit (highest return) for Farmer Jane.

##### 2b. Solve in a different method.

In [1]:
# the types of produce
product = [:wheat, :corn]

# fertilizer required for each type of prodcut
fert   = Dict( :wheat => 2, :corn => 4)

# labor required for each type of product
labor = Dict( :wheat => 3, :corn => 2)

# acres required for each crop --- not sure if I actually need this, do not think so ::TODO::
acres = Dict( :wheat => 1, :corn => 1)

# profit made for each product
profit = Dict( :wheat => 230, :corn => 190)

# quantities in stock for each ingredient
num_fert     = 130
num_labor    = 100
num_acres    = 45
;

In [2]:
using JuMP, Clp
m = Model(solver = ClpSolver())

@variable(m, crop[product] >= 0 )    # "crop" is a dictionary indexed over product

@expression(m, tot_labor,   sum(crop[i] * labor[i]   for i in product) )
@expression(m, tot_fert,    sum(crop[i] * fert[i]    for i in product) )
@expression(m, tot_acres,   sum(crop[i] * acres[i]    for i in product) )
@expression(m, tot_profit,  sum(crop[i] * profit[i]  for i in product) )

@constraint(m, crop[:wheat] <= num_acres )      # maximum number of wheat that can be produced 
@constraint(m, crop[:corn]  <= num_acres )      # maximum number of corn that can be produced 
@constraint(m, tot_labor    <= num_labor )      # maximum number of labor needed
@constraint(m, tot_fert     <= num_fert )       # maximum number of fertilizer needed

@objective(m, Max, tot_profit)

solve(m)
println(getvalue(crop))
println("Total profit will be \$", getvalue(tot_profit))
println("Fertilizer needed: ", getvalue(tot_fert))
println("Acres used: ", getvalue(tot_acres))
println("People needed: ", getvalue(tot_labor))

crop: 1 dimensions:
[wheat] = 17.499999999999993
[ corn] = 23.750000000000004
Total profit will be $8537.5
Fertilizer needed: 130.0
Acres used: 41.25
People needed: 99.99999999999999


##### As we can see here, the two values are identical. Therefore, I feel confident saying that I have found the correct answer. The recommendation remains to cover 17.5 acres with Wheat, and 23.75 acres of Corn. 